### DISCUSSION TITLE
Need help with create_model statement

### DISCUSSION THREAD

**Discussion by shanedp** [pydantic/pydantic#11132] [Question]
I can't seem to find documentation on the proper syntax for creating a model using create_model. I need to create a pydantic model dynamically, so the classic and well-documented example using a BaseModel class won't work. 

I've not seen any documentation anywhere that shows how I would use create_model to build a model for nested structures. Could someone provide create_model code for the following dict?

`{'a': 123, 'nested': {'c': 'abc', 'd': 456}, 'list': [123, 'abc']}`

**Comment by Viicos** [comment]
See https://docs.pydantic.dev/dev/concepts/models/#dynamic-model-creation.

For 2.10, see https://docs.pydantic.dev/latest/concepts/models/#dynamic-model-creation. The field specification format changed a bit (and the documentation as well) on `main`.

**Reply by shanedp** [reply]
@Viicos thank you for your response and for sending a link to your dynamic model creation documentation. I see it shows simple use cases for create_model() and discusses what is to be included in the arguments passed. 

For example, if I want to create a model for a simple dict such as `{'A': 'a'}` I could use it as follows:
`model = create_model('my_model', A=(str,'a'))`

However, if I want to make a slightly more complicated model using a dict like `{'A': 'a', 'nested': {'B': 'b', 'C': 'c'}}` I'm not sure what the syntax would be. Could you give a working example for this dictionary?

**Reply by shanedp** [reply]
A bit of further info, the closest I've been able to find this done anywhere is from [this SO post](https://stackoverflow.com/questions/69216961/how-to-create-dynamic-models-using-pydantic-and-a-dict-data-type/69225701#69225701), where it was demonstrated that an Pydantic model could be built using create_model() and a recursive function and based on an example dictionary:

```py
from typing import Dict
from pydantic import create_model

test_dict = {'a': 12, 'nested': {'b': 'value', 'c': 1}}


def _make_model(v, name):
    if type(v) is dict:
        return create_model(name, **{k: _make_model(v, k) for k, v in v.items()}), ...
    return type(v), v


def make_model(v: Dict, name: str):
    return _make_model(v, name)[0]


model = make_model(test_dict, 'MyModel')
m = model.parse_obj(test_dict)
print(m)
print(model.schema_json(indent=2))
```

The function `_make_model` recursively traverses `test_dict`, but it's not clear to me how the `create_model()` call is built out. I would use this example but there is no coverage for lists or lists of dicts. 

I would be very appreciative if you could provide working code for how one can construct a `create_model()` call for these use cases.

**Reply by Viicos** [reply]
It's no clear how `{'a': 12, 'nested': {'b': 'value', 'c': 1}}` should be converted to a Pydantic model. Is this your expected syntax?

`{field_key: field_default, ...}`

What about fields without a default value?
